### ISSUE TITLE
"code": 422 AS3 pointer does not point to required object type

### ISSUE THREAD

**Issue by MMarco1977** [F5Networks/f5-appsvcs-extension#475]
Environment
 * Application Services Version: 3.28.0
 * BIG-IP Version: 15.1.3-0.0.11

### Summary
After converting the json with ACC, the POST AS3 declaration fails

Steps To Reproduce
1) Imported UCS file into Visual Studio Code and explored TMOS config
2) Converted to ACC
3) Insert / Remove schema reference
4) Post as AS3 Declaration

1. Submit the following declaration:
```json
{
    "$schema": "https://raw.githubusercontent.com/F5Networks/f5-appsvcs-extension/master/schema/latest/as3-schema.json",
    "class": "AS3",
    "declaration": {
        "class": "ADC",
        "schemaVersion": "3.28.0",
        "id": "urn:uuid:d7c2c3db-7fc8-42f1-ba08-9b5e5a3d1332",
        "label": "Converted Declaration",
        "remark": "Auto-generated by AS3 Config Converter",
        "Common": {
            "class": "Tenant",
            "Shared": {
                "class": "Application",
                "template": "shared",
                "voip": {
                    "remark": "voip",
                    "class": "GSLB_Data_Center"
                },
                "ACM-AN1": {
                    "dataCenter": {
                        "use": "/Common/Shared/voip"
                    },
                    "devices": [
                        {
                            "address": "1.6.99.166"
                        }
                    ],
                    "serverType": "generic-host",
                    "virtualServers": [
                        {
                            "address": "1.6.99.166",
                            "port": 0,
                            "name": "ACM-AN1_VS"
                        }
                    ],
                    "class": "GSLB_Server"
                },
                "ACM-BA1": {
                    "dataCenter": {
                        "use": "/Common/Shared/voip"
                    },
                    "devices": [
                        {
                            "address": "1.6.115.166"
                        }
                    ],
                    "serverType": "generic-host",
                    "virtualServers": [
                        {
                            "address": "1.6.115.166",
                            "port": 0,
                            "name": "ACM-BA1_VS"
                        }
                    ],
                    "class": "GSLB_Server"
                },
                "pool_ACM-AN1": {
                    "lbModeAlternate": "none",
                    "lbModeFallback": "none",
                    "members": [
                        {
                            "ratio": 1,
                            "server": {
                                "use": "/Common/Shared/ACM-AN1"
                            },
                            "virtualServer": "ACM-AN1_VS"
                        }
                    ],
                    "verifyMemberEnabled": false,
                    "class": "GSLB_Pool",
                    "resourceRecordType": "A"
                },
                "pool_ACM-BA1": {
                    "lbModeAlternate": "none",
                    "lbModeFallback": "none",
                    "members": [
                        {
                            "ratio": 1,
                            "server": {
                                "use": "/Common/Shared/ACM-BA1"
                            },
                            "virtualServer": "ACM-BA1_VS"
                        }
                    ],
                    "verifyMemberEnabled": false,
                    "class": "GSLB_Pool",
                    "resourceRecordType": "A"
                },
                "pool_nomatc": {
                    "lbModeAlternate": "none",
                    "fallbackIP": "127.0.0.2",
                    "lbModeFallback": "drop-packet",
                    "lbModePreferred": "fallback-ip",
                    "class": "GSLB_Pool",
                    "resourceRecordType": "A"
                },
                "voip.it": {
                    "poolLbMode": "topology",
                    "pools": [
                        {
                            "use": "/Common/Shared/pool_ACM-AN1"
                        },
                        {
                            "use": "/Common/Shared/pool_ACM-BA1"
                        }
                    ],
                    "class": "GSLB_Domain",
                    "resourceRecordType": "A",
                    "domainName": "voip.it"
                },
                "topology": {
                    "records": [
                        {
                            "destination": {
                                "matchType": "pool",
                                "matchOperator": "equals",
                                "matchValue" : {"use": "pool_ACM-AN1"}
                            },
                            "source": {
                                "matchType": "subnet",
                                "matchOperator": "equals",
                                "matchValue": "10.103.216.18/32"
                            },
                            "weight": 5
                        },
                        {
                            "destination": {
                                "matchType": "pool",
                                "matchOperator": "equals",
                                "matchValue" : {"use": "pool_ACM-BA1"}
                            },
                            "source": {
                                "matchType": "subnet",
                                "matchOperator": "equals",
                                "matchValue": "10.103.216.2/32"
                            },
                            "weight": 5
                        },
                        {
                            "destination": {
                                "matchType": "pool",
                                "matchOperator": "equals",
                                "matchValue" : {"use": "pool_nomatc"}
                            },
                            "source": {
                                "matchType": "subnet",
                                "matchOperator": "equals",
                                "matchValue": "0.0.0.0/0"
                            },
                            "weight": 5
                        }
                    ],
                    "class": "GSLB_Topology_Records",
                    "longestMatchEnabled": false
                }
            }
        }
    }
}
```
2. Observe the following error response:
```json
{
 "id": "860b22cc-6ba3-409f-b90a-19a1f3ef6433",
 "results": [
 {
 "code": 422,
 "errors": [
 "/Common/Shared/topology/records/0/destination/matchValue/use: AS3 pointer pool_ACM-AN1 does not point to required object type"
 ],
 "declarationFullId": "",
 "message": "declaration is invalid"
 }
 ],
 "declaration": {}
}
```

3. Expected Behavior:
```
gtm datacenter /Common/voip {
    description voip
}
gtm server /Common/ACM-AN1 {
    datacenter /Common/voip
    devices {
        ACME-AN1 {
            addresses {
                1.6.99.166 { }
            }
        }
    }
    product generic-host
    virtual-servers {
        ACM-AN1_VS {
            destination 1.6.99.166:0
        }
    }
}


gtm topology ldns: subnet 10.103.216.18/32 server: pool /Common/pool_ACM-AN1 {
    order 1
    score 5
}


ltm wideip a /Common/voip.it {
    pool-lb-mode topology
    pools {
        /Common/pool_ACM-AN1 {
            order 3
        }

```

**Comment by dstokesf5** [comment]
Thank you for reporting this issue. I have confirmed the bug and added it to our internal product backlog as AUTOTOOL-2590.

**Comment by dstokesf5** [comment]
This issue has been resolved with the 3.30.0 release of AS3.
